src/snowflake/snowpark/modin/pandas/indexing.py

```diff
@@ -1039,6 +1039,9 @@ def __setitem__(
         )
         if item_is_2d_array:
             item = pd.DataFrame(item)
+        frame_is_df_and_item_is_series = isinstance(item, pd.Series) and isinstance(
```

Comment:
what happens for item is a Index or list? Are they matching with pandas? Can you verify too?

Reply:
Added tests!